src/main/PDF/Services/CrudServices/UploadPDFService.java

```diff
@@ -0,0 +1,193 @@
+  }
+
+  @Override
+  public Message executeAndGetResponse() {
+    if (pdfType == null) {
+      return PdfMessage.INVALID_PDF_TYPE;
+    } else if (fileStream == null) {
+      return PdfMessage.INVALID_PDF;
+    } else if (!fileContentType.equals("application/pdf")
+        && !fileContentType.equals("application/octet-stream")
+        && !fileContentType.startsWith("image")) {
+      return PdfMessage.INVALID_PDF;
+    } else {
+      if (fileContentType.startsWith("image")) {
+        try {
+          fileStream = convertImageToPDF(fileStream);
+          filename = filename.substring(0, filename.lastIndexOf(".")) + ".pdf";
+        } catch (IOException exception) {
+          return PdfMessage.INVALID_PDF;
+        }
+      }
+      if ((pdfType == PDFType.COMPLETED_APPLICATION
+              || pdfType == PDFType.IDENTIFICATION_DOCUMENT
+              || pdfType == PDFType.BLANK_FORM)
+          && (privilegeLevel == UserType.Client
+              || privilegeLevel == UserType.Worker
+              || privilegeLevel == UserType.Director
+              || privilegeLevel == UserType.Admin
+              || privilegeLevel == UserType.Developer)) {
+        try {
+          return mongodbUpload();
+        } catch (GeneralSecurityException | IOException e) {
+          return PdfMessage.SERVER_ERROR;
+        }
+      } else {
+        return PdfMessage.INSUFFICIENT_PRIVILEGE;
+      }
+    }
+  }
+
+  public Message mongodbUpload() throws GeneralSecurityException, IOException {
+    String title = PdfController.getPDFTitle(filename, fileStream, pdfType);
+    GridFSBucket gridBucket = GridFSBuckets.create(db, pdfType.toString());
+    GridFSUploadOptions options;
+    InputStream inputStream;
+
+    if (pdfType == PDFType.BLANK_FORM) {
+      inputStream = fileStream;
+      options =
+          new GridFSUploadOptions()
+              .chunkSizeBytes(CHUNK_SIZE_BYTES)
+              .metadata(
+                  new Document("type", "pdf")
+                      .append("upload_date", String.valueOf(LocalDate.now()))
+                      .append("title", title)
+                      .append("annotated", false)
+                      .append("uploader", uploader)
+                      .append("organizationName", organizationName));
+
+    } else {
+      inputStream = encryptionController.encryptFile(fileStream, uploader);
+      options =
+          new GridFSUploadOptions()
+              .chunkSizeBytes(CHUNK_SIZE_BYTES)
+              .metadata(
+                  new Document("type", "pdf")
+                      .append("upload_date", String.valueOf(LocalDate.now()))
+                      .append("title", title)
+                      .append("uploader", uploader)
+                      .append("organizationName", organizationName));
+    }
+    gridBucket.uploadFromStream(filename, inputStream, options);
+    return PdfMessage.SUCCESS;
+  }
+
+  public InputStream convertImageToPDF(InputStream fileStream) throws IOException {
```

Comment:
can we move all the image upload code to a separate PR and remove it from this PR